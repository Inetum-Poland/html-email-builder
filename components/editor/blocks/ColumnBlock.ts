import { useI18n } from "vue-i18n";
import type { BlockProperties } from "grapesjs";
import { Square, Columns2, Columns3, Columns4 } from "lucide-static";

const createContent = (count: number) => {
  const { t } = useI18n();

  return `
  <mj-section>
    ${`<mj-column><mj-text>${t("insertTextHere")}</mj-text></mj-column>`.repeat(count)}
  </mj-section>
  `;
};

const icons = new Map([
  [1, Square],
  [2, Columns2],
  [3, Columns3],
  [4, Columns4],
]);

export const ColumnBlock = (count: 1 | 2 | 3 | 4): BlockProperties => {
  const { t } = useI18n();

  return {
    select: true,
    label: t("column", count),
    media: icons.get(count),
    content: createContent(count),
  };
};
