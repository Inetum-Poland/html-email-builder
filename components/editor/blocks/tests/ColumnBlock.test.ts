import { expect, test } from "vitest";
import { ColumnBlock } from "../ColumnBlock";

test.each([1, 2, 3] as const)(
  "renders the Column block with %i column(s)",
  (count) => expect(ColumnBlock(count)).toMatchSnapshot()
);
