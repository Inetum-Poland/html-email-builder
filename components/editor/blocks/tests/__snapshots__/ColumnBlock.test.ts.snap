// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the Column block with 1 column(s) 1`] = `
{
  "content": "
  <mj-section>
    <mj-column><mj-text>Insert text here...</mj-text></mj-column>
  </mj-section>
  ",
  "label": "1 column",
  "media": "
<svg
  class="lucide lucide-square"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect width="18" height="18" x="3" y="3" rx="2" />
</svg>
",
  "select": true,
}
`;

exports[`renders the Column block with 2 column(s) 1`] = `
{
  "content": "
  <mj-section>
    <mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column>
  </mj-section>
  ",
  "label": "2 columns",
  "media": "
<svg
  class="lucide lucide-columns-2"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect width="18" height="18" x="3" y="3" rx="2" />
  <path d="M12 3v18" />
</svg>
",
  "select": true,
}
`;

exports[`renders the Column block with 3 column(s) 1`] = `
{
  "content": "
  <mj-section>
    <mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column>
  </mj-section>
  ",
  "label": "3 columns",
  "media": "
<svg
  class="lucide lucide-columns-3"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect width="18" height="18" x="3" y="3" rx="2" />
  <path d="M9 3v18" />
  <path d="M15 3v18" />
</svg>
",
  "select": true,
}
`;

exports[`renders the Column block with 4 column(s) 1`] = `
{
  "content": "
  <mj-section>
    <mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column><mj-column><mj-text>Insert text here...</mj-text></mj-column>
  </mj-section>
  ",
  "label": "4 columns",
  "media": "
<svg
  class="lucide lucide-columns-4"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect width="18" height="18" x="3" y="3" rx="2" />
  <path d="M7.5 3v18" />
  <path d="M12 3v18" />
  <path d="M16.5 3v18" />
</svg>
",
  "select": true,
}
`;
